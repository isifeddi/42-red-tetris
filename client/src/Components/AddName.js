import { useState } from "react";
import { addPlayerName } from "../actions/playerAction";
import { useDispatch } from "react-redux";
import { StyledButton, StyledInput, StyledAddnameWrapper } from "./styling/StyledForm";

const AddName = () => {
  const dispatch = useDispatch();
  const [username, setUserName] = useState("");

  function adduser() {
    dispatch(addPlayerName(username));
  }
  return (
    <StyledAddnameWrapper>
      <StyledInput
        placeholder="Username"
        onChange={(e) => setUserName(e.target.value)}
      />
      <StyledButton onClick={() => adduser()}>Submit</StyledButton>
    </StyledAddnameWrapper>
  );
};

export default AddName;
