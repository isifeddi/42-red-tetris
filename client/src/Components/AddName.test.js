import renderer from "react-test-renderer";
import AddName from "./AddName";
import store from "../Store";
import { Provider } from "react-redux";
test("Add Name render test ", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <AddName />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
test("Add Name Mock ", () => {
  const tree = renderer
    .create(
      <Provider store={store}>
        <AddName />
      </Provider>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});