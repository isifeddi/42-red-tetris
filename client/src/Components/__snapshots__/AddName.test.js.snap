// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Add Name render test  1`] = `
<form
  className="sc-dlfnbm dOZfKl"
  onSubmit={[Function]}
>
  <input
    className="sc-bdfBwQ eNvKjX"
    id="username-i"
    onChange={[Function]}
    placeholder="Enter Username"
  />
  <button
    className="sc-gsTCUz fKrtQw"
  >
    Submit
  </button>
</form>
`;

exports[`App render test  1`] = `
<form
  className="sc-dlfnbm dOZfKl"
  onSubmit={[Function]}
>
  <input
    className="sc-bdfBwQ eNvKjX"
    id="username-i"
    onChange={[Function]}
    placeholder="Enter Username"
  />
  <button
    className="sc-gsTCUz fKrtQw"
  >
    Submit
  </button>
</form>
`;
